libs/features/mas/commerce/src/checkout-link.js

```diff
@@ -185,8 +186,18 @@ export class HTMLCheckoutAnchorElement extends HTMLAnchorElement {
             if (text) this.firstElementChild.innerHTML = text;
             if (className) this.classList.add(...className.split(' '));
             if (handler) {
-                this.setAttribute('href', '#');
+              this.setAttribute('href', '#');
+              if (this.getAttribute('data-modal') === 'true') {
+                const hash = `${offers.flat()[0].offerType === "TRIAL" ? 'try' : 'buy'}-${this.closest('merch-card')?.getAttribute('name')}`;
+                this.setAttribute('hash', hash);
+                this.#checkoutActionHandler = handler.bind(this);
+                if (window.location.hash.replace('#', '') === hash && !this.#modalWasOpened) {
```

Comment:
it feels look like this logic belongs to [openModal](https://github.com/adobecom/milo/blob/stage/libs/blocks/merch/merch.js#L360) function defined in merch.js.

checkout-link custom element cannot be too opinionated how things should be, since it can be used anywhere outside Milo.

Reply:
@yesil It works this way: when a user opens a modal, goes to a different page, then goes back - on page load we need to initiate a click on the CTA with a respective hash.
The `openModal` method is not called at page load.
Alternatively, I can set up an on-page load event listener and on each page load check if there is a CTA with the hash that matches the hash in the page URL, and then click on it, but I think having this check on each page load would be redundant, or should I do it?

Reply:
I think this logic could be in the merch.js, once initialized, if there is a hash, then we can perform this check.
It should be executed only on back navigation.